File: Notes/Computer Science/Notebook/python/phi.py
phi=[0]*10000001
for a in range(0,1+10000000):
	phi[a]=a
for a in range(2,1+10000000):
	if phi[a]==a:
		for b in range (a,1+10000000,a):
			phi[b]=phi[b]-phi[b]//a

File: Notes/Computer Science/Notebook/number theory/phi.cpp
// Description: phi function is defined as phi(n)=# of positive integers <=n that are coprime with n
const I lim=5000001;
I phi[lim];
Z calculatephi(){//Håkan Terelius
	rep(i,0,lim)phi[i]=i&1?i:i/2;
	for(I i=3;i<lim;i+=2)if(phi[i]==i)
		for(I j=i;j<lim;j+=i)phi[j]-=phi[j]/i;
}